### PR TITLE
US012: Create Register Form

### PR DESCRIPTION
**Pull request by chungdaotn**

### CODE REVIEW

File: B-CLIENT/gotrip/src/App.js [hunk 1]
```diff
@@ -2,6 +2,7 @@ import logo from './logo.svg';
 import './App.css';
 import { BrowserRouter, Switch, Route, Link, Router } from 'react-router-dom';
 import AdminPage from './admin-page/AdminPage';
+import register from'./auth/registration/register'
```
Comment (dnhieu1992) [inline B-CLIENT/gotrip/src/App.js:5]: Name of component is PascalCase(Register)

File: B-CLIENT/gotrip/src/App.js [hunk 2]
```diff
@@ -11,6 +12,7 @@ function App() {
         {/* <Route exact path="/" component={LoginContainer} exact/> */}
         <Route path="/" exact component={AdminPage} />
         <Route path="/admin" component={AdminPage} />
+        <Route path ="/registration" component={register}/>
```
Comment (dnhieu1992) [inline B-CLIENT/gotrip/src/App.js:15]: path="/register"

File: B-CLIENT/gotrip/src/auth/registration/register.jsx [hunk 3]
```diff
@@ -0,0 +1,169 @@
+import Formsy, { addValidationRule } from "formsy-react";
+import { useState } from "react";
+import { useHistory } from "react-router";
+import { FormsyElement, Input, LoaderButton } from "../../shared/components";
+import { REGISTER_TEXT_CONFIG } from "./constants/resources";
+
+const register = () => {
```
Comment (dnhieu1992) [inline B-CLIENT/gotrip/src/auth/registration/register.jsx:7]: Name of Component is PascalCase

File: B-CLIENT/gotrip/src/auth/registration/register.jsx [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+    const [isValid, setIsValid] = useState(true);
+    const history = useHistory();
+
+    const { FormsyInput } = FormsyElement;
+
+    const {
+        username,
+        email,
+        password,
+        confirm_password
```
Comment (dnhieu1992) [inline B-CLIENT/gotrip/src/auth/registration/register.jsx:19]: rename confim_password to confirmPassword

File: B-CLIENT/gotrip/src/auth/registration/register.jsx [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+                            placeholder: "Enter your password"
+                        }}
+                        name="password"
+                        label="Password"
+                        value={password}
+                        required
+                        validationError={REGISTER_TEXT_CONFIG.PASSWORD_REQUIRED_MSG}
+                        validations={{ matchRegexp: /^(?=.*[~!@#$%^&*?()])(?=.*[a-z])(?=.*[A-Z])/, minLength: 8, maxLength: 30 }}
+                        validationErrors={{
+                            matchRegexp: REGISTER_TEXT_CONFIG.PASSWORD_TYPE,
```
Comment (dnhieu1992) [inline B-CLIENT/gotrip/src/auth/registration/register.jsx:118]: change PASSWORD_INVALID_MSG

File: B-CLIENT/gotrip/src/auth/registration/register.jsx [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+                            maxLength: REGISTER_TEXT_CONFIG.PASSWORD_MAX_LENGTH
+                        }}
+                    />
+                </div>
+                <div className="form__input">
+                    <FormsyInput
+                        inputProps={{
+                            type: "password",
+                            className: "form-input",
+                            placeholder: "Enter your confirm password"
```
Comment (dnhieu1992) [inline B-CLIENT/gotrip/src/auth/registration/register.jsx:129]: move to constant

File: B-CLIENT/gotrip/src/shared/sass/register/register.scss [hunk 7]
```diff
@@ -0,0 +1,77 @@
+// *{
```
Comment (dnhieu1992) [inline B-CLIENT/gotrip/src/shared/sass/register/register.scss:1]: remove code not use

File: B-CLIENT/gotrip/src/shared/sass/register/register.scss [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+//     box-sizing: border-box;
+//     font-family: Arial, Helvetica, sans-serif;
+//     word-spacing: 6px;
+//     font-weight: 100;
+//     font-size: small;
+// }
+.main {
+    width: 100%;
+    height: 700px;
+    background-image: url('https://1.bp.blogspot.com/-RZWWWvj_DGI/XnC16WAl0hI/AAAAAAAAY2s/DABOYajTivYI90bU8qEegEAuT6UAp-LTQCLcBGAsYHQ/s1600/Rung-nui-hung-vy-Diembaogi%2B%252812%2529.jpg');
```
Comment (dnhieu1992) [inline B-CLIENT/gotrip/src/shared/sass/register/register.scss:14]: download image to local and import to Upload folder

File: B-CLIENT/gotrip/src/shared/sass/register/register.scss [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+    background-size: cover;
+    background-repeat: no-repeat;
+    position: relative;
+
+    & h1{
```
Comment (dnhieu1992) [inline B-CLIENT/gotrip/src/shared/sass/register/register.scss:19]: invalid code, with tag no need &

File: B-CLIENT/gotrip/src/auth/registration/register.jsx [hunk 10]
```diff
@@ -0,0 +1,169 @@
...
+                            type: "password",
+                            className: "form-input",
+                            placeholder: "Enter your password"
+                        }}
+                        name="password"
+                        label="Password"
+                        value={password}
+                        required
+                        validationError={REGISTER_TEXT_CONFIG.PASSWORD_REQUIRED_MSG}
+                        validations={{ matchRegexp: /^(?=.*[~!@#$%^&*?()])(?=.*[a-z])(?=.*[A-Z])/, minLength: 8, maxLength: 30 }}
```
Comment (dnhieu1992) [inline B-CLIENT/gotrip/src/auth/registration/register.jsx:116]: move regular expression to constant

File: B-CLIENT/gotrip/src/App.js (hunk 1 above)
Comment (chungdaotn) [reply B-CLIENT/gotrip/src/App.js]: done

File: B-CLIENT/gotrip/src/App.js (hunk 2 above)
Comment (chungdaotn) [reply B-CLIENT/gotrip/src/App.js]: done

File: B-CLIENT/gotrip/src/auth/registration/register.jsx (hunk 3 above)
Comment (chungdaotn) [reply B-CLIENT/gotrip/src/auth/registration/register.jsx]: done

File: B-CLIENT/gotrip/src/auth/registration/register.jsx (hunk 4 above)
Comment (chungdaotn) [reply B-CLIENT/gotrip/src/auth/registration/register.jsx]: done

File: B-CLIENT/gotrip/src/auth/registration/register.jsx (hunk 5 above)
Comment (chungdaotn) [reply B-CLIENT/gotrip/src/auth/registration/register.jsx]: done

File: B-CLIENT/gotrip/src/auth/registration/register.jsx (hunk 6 above)
Comment (chungdaotn) [reply B-CLIENT/gotrip/src/auth/registration/register.jsx]: done

File: B-CLIENT/gotrip/src/shared/sass/register/register.scss (hunk 7 above)
Comment (chungdaotn) [reply B-CLIENT/gotrip/src/shared/sass/register/register.scss]: done

File: B-CLIENT/gotrip/src/shared/sass/register/register.scss (hunk 8 above)
Comment (chungdaotn) [reply B-CLIENT/gotrip/src/shared/sass/register/register.scss]: done

File: B-CLIENT/gotrip/src/shared/sass/register/register.scss (hunk 9 above)
Comment (chungdaotn) [reply B-CLIENT/gotrip/src/shared/sass/register/register.scss]: done

File: B-CLIENT/gotrip/src/shared/sass/register/register.scss (hunk 8 above)
Comment (dnhieu1992) [reply B-CLIENT/gotrip/src/shared/sass/register/register.scss]: move to upload folder

File: B-CLIENT/gotrip/src/auth/registration/register.jsx [hunk 11]
```diff
@@ -0,0 +1,169 @@
+import Formsy, { addValidationRule } from "formsy-react";
+import { useState } from "react";
+import { useHistory } from "react-router";
+import { FormsyElement, Input, LoaderButton } from "../../shared/components";
+import { REGISTER_TEXT_CONFIG } from "./constants/resources";
+
+const Register = () => {
```
Comment (dnhieu1992) [inline B-CLIENT/gotrip/src/auth/registration/register.jsx:7]: name of file same with name of component (PasscalCase)

File: B-CLIENT/gotrip/src/auth/registration/register.jsx (hunk 10 above)
Comment (chungdaotn) [reply B-CLIENT/gotrip/src/auth/registration/register.jsx:116]: done

File: B-CLIENT/gotrip/src/shared/sass/register/register.scss (hunk 8 above)
Comment (chungdaotn) [reply B-CLIENT/gotrip/src/shared/sass/register/register.scss]: done

File: B-CLIENT/gotrip/src/auth/registration/register.jsx (hunk 11 above)
Comment (chungdaotn) [reply B-CLIENT/gotrip/src/auth/registration/register.jsx:7]: done

**Pull request merged**